### PR TITLE
Feature/book searcher

### PR DESCRIPTION
**Pull request by smichaelshal**

### CODE REVIEW

File: src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+        this.storage = mock(Storage.class);
+        this.filter = mock(Function.class);
+        this.searcher = new BookSearcher(this.storage);
+    }
+
+    @Test
+    void search() {
+        // Arrange
+        List<Book> expectedResult = new LinkedList();
+        when(this.storage.readAll(any(Function.class))).thenReturn(expectedResult);
```
Comment (Iris2534) [inline src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java:33]: Don't use `any`, use your mock filter

File: src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        // Act
+        List<Book> searchResult = this.searcher.search(this.filter);
+
+        // Assert
+        assertEquals(searchResult, expectedResult);
```
Comment (Iris2534) [inline src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java:39]: Is this the correct order of parameters?

File: src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java (hunk 1 above)
Comment (smichaelshal) [reply src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java:33]: 1. The denominator is to use `BookIdMatcher`?
2. Why do we need the mock filter if for every result I expect I already know what I want to get?

File: src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java (hunk 2 above)
Comment (smichaelshal) [reply src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java:39]: fix

File: src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java (hunk 1 above)
Comment (Iris2534) [reply src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java]: You know what you want to get, but you also want to test that the storage will be called with the specific filter you gave to it, and not some random filter.

Comment (smichaelshal) [reply src/test/java/org/plasma/digitalib/searchers/BookSearcherTest.java]: fix

**Comment by Iris2534** [comment]
Approved. Merge when the branches before will be merged

**Pull request merged**